src/databricks/labs/remorph/reconcile/schema_compare.py

```diff
@@ -0,0 +1,131 @@
+from pyspark.sql.types import BooleanType, StringType, StructField, StructType
+from sqlglot import parse_one
+
+from databricks.labs.remorph.config import MorphConfig
+from databricks.labs.remorph.reconcile.constants import SourceType
+from databricks.labs.remorph.reconcile.recon_config import (
+    Schema,
+    SchemaMatchResult,
+    SchemCompareOutput,
+    Table,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SchemaCompare:
+    def __init__(
+        self,
+        spark: SparkSession,
+    ):
+        self.spark = spark
+
+    # Define the schema for the schema compare DataFrame
+    _schema_compare_schema: StructType = StructType(
+        [
+            StructField("source_column", StringType(), False),
+            StructField("source_datatype", StringType(), False),
+            StructField("databricks_column", StringType(), True),
+            StructField("databricks_datatype", StringType(), True),
+            StructField("is_valid", BooleanType(), False),
+        ]
+    )
+
+    @classmethod
+    def _build_master_schema(
+        cls,
+        source_schema: list[Schema],
+        databricks_schema: list[Schema],
+        table_conf: Table,
+    ) -> list[SchemaMatchResult]:
+        master_schema = source_schema
+        if table_conf.select_columns:
+            master_schema = [s for s in master_schema if s.column_name in table_conf.select_columns]
+        if table_conf.drop_columns:
+            master_schema = [s for s in master_schema if s.column_name not in table_conf.drop_columns]
+
+        target_column_map = table_conf.to_src_col_map or {}
+        master_schema = [
+            SchemaMatchResult(
+                source_column=s.column_name,
+                databricks_column=target_column_map.get(s.column_name, s.column_name),
+                source_datatype=s.data_type,
+                databricks_datatype=next(
+                    (
+                        tgt.data_type
+                        for tgt in databricks_schema
+                        if tgt.column_name == target_column_map.get(s.column_name, s.column_name)
+                    ),
+                    "",
+                ),
+            )
+            for s in master_schema
+        ]
+        return master_schema
+
+    def _create_dataframe(self, data: list, schema: StructType) -> DataFrame:
+        """
+        :param data: Expectation is list of dataclass
+        :param schema: Target schema
+        :return: DataFrame
+        """
+        data = [tuple(asdict(item).values()) for item in data]
+        df = self.spark.createDataFrame(data, schema)
+
+        return df
+
+    @classmethod
+    def _parse(cls, source: str, column: str, data_type: str) -> str:
+        config = MorphConfig(source=source)
+        return (
+            parse_one(f"create table dummy ({column} {data_type})", read=config.get_read_dialect())
+            .sql(dialect=config.get_write_dialect())
+            .replace(", ", ",")
```

Comment:
why do you need this replace?

Reply:
Parsing adds space, suppose like DECIMAL(10, 20), but when you consume schema from information schema/describe command, the output is like DECIMAL(10,20) .. to handle that, I am doing this.